### PR TITLE
chore(batch): deprecate chunk size in scan

### PR DESCRIPTION
**Pull request by chenzl25**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- As we have `limit` which can be used to decide the chunk size of the scan, we can deprecate the `chunk_size` field in proto.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: proto/batch_plan.proto [hunk 1]
```diff
@@ -25,11 +25,6 @@ message RowSeqScanNode {
   // Whether the order on output columns should be preserved.
   bool ordered = 5;
 
-  message ChunkSize {
-    uint32 chunk_size = 1;
-  }
-  // If along with `batch_limit`, `chunk_size` will be set.
-  ChunkSize chunk_size = 6;
```
Comment (kwannoel) [inline proto/batch_plan.proto:32]: Shall we just mark it as `deprecated`? So others won't accidentally use `6` next time.

**Review by kwannoel — APPROVED** [review]

**Comment by kwannoel** [comment]
By `limit` you mean `rate_limit`? Previously I suppose we use `chunk_size` for that purpose (kind of)?

**Comment by chenzl25** [comment]
> By `limit` you mean `rate_limit`? Previously I suppose we use `chunk_size` for that purpose (kind of)?

This limit refers to the sql `limit`. For example `select * from t limit 1` limit 1 can be pushed into scan.

File: proto/batch_plan.proto [hunk 2]
```diff
@@ -25,11 +25,10 @@ message RowSeqScanNode {
...
+  // deprecated
+  // ChunkSize chunk_size = 6;
+  // reserved 6;
+
```
Comment (kwannoel) [inline proto/batch_plan.proto:31]:
```suggestion
  message ChunkSize {
    uint32 chunk_size = 1;
  }
  ChunkSize chunk_size = 6 [deprecated = true];
```

Comment (kwannoel) [reply proto/batch_plan.proto:31]: Sorry I should have left the suggestion directly 😆 I was referring to `deprecated` syntax of protobuf.

File: proto/batch_plan.proto (hunk 1 above)
Comment (BugenZhao) [reply proto/batch_plan.proto:32]: I guess simply removing it is okay. We never persist batch plans.

Comment (chenzl25) [reply proto/batch_plan.proto:32]: I guess we can simply use `limit = 6` to replace `chunk_size =6`.

**Pull request merged**